src/compiler/checker.ts

```diff
@@ -26522,6 +26523,31 @@ namespace ts {
             let prop: Symbol | undefined;
             if (isPrivateIdentifier(right)) {
                 const lexicallyScopedSymbol = lookupSymbolForPrivateIdentifierDeclaration(right.escapedText, right);
+
+                if (lexicallyScopedSymbol && (compilerOptions.target === ScriptTarget.ESNext && !useDefineForClassFields)) {
+                    const lexicalValueDecl = lexicallyScopedSymbol.valueDeclaration;
+                    const lexicalClass = getContainingClass(lexicalValueDecl);
+                    const parentStaticFieldInitializer = findAncestor(node, (n) => {
+                        if (n === lexicalClass) return "quit";
+                        if (isPropertyDeclaration(n.parent) && n.parent.initializer === n && n.parent.parent === lexicalClass) {
+                            return true;
+                        }
+                        return false;
+                    });

```

Comment:
```suggestion
                    const parentStaticPropertyInitializer = findAncestor(node, (n) => n === lexicalClass 
                        ? "quit"
                        : isPropertyDeclaration(n.parent) && n.parent.initializer === n && n.parent.parent === lexicalClass);
```

Reply:
I'm confused, doesn't this code have to check that the property being initialised is `static`. I don't see that check here, but it's not an error if it's an instance property, right?

Reply:
Related: Is there an existing test for this case? The new test covers the private/static case but not the private/instance case.

Reply:
> I'm confused, doesn't this code have to check that the property being initialised is `static`. I don't see that check here, but it's not an error if it's an instance property, right?

Yeah it should... fail on my part. Fixed now. Also added tests to validate the allowed instance cases